lib/rules.js

```diff
@@ -353,6 +353,267 @@ export const GROUP_TERMS_BY_ROOT = defineRule(
+
+                const secondCoef = coefficients[1][0]
+                // Second polynomial to be matched
+                const firstMiddle = firstPoly.map(term => {
+                    const identifier = term.args[0]
+                    const exponent = build.number(query.getValue(term.args[1]) / 2)
+                    return build.applyNode('pow', [identifier, exponent])
+                })
+
+                let secondMiddle
+                if (thirdPoly) {
+                    secondMiddle = thirdPoly.map(term => {
+                        const identifier = term.args[0]
+                        const exponent = build.number(query.getValue(term.args[1]) / 2)
+                        return build.applyNode('pow', [identifier, exponent])
+                    })
+                }
+
+                const matchSecondTerm =  secondMiddle
+                      ? build.applyNode ('mul',
+                                         [secondCoef, ...firstMiddle, ...secondMiddle]
+                                         , {implicit: true})
+                      : build.applyNode ('mul',
+                                         [secondCoef, ...firstMiddle]
+                                         , {implicit: true})
+
+                // General conditions:
+                // All degrees in 1st and 3rd term must be even
+                // secondTerm = matchSecondTerm
+                isFactorable =
+                    firstPoly.forEach(term => {return term.args[1] % 2 == 0})
+                    && thirdPoly ? thirdPoly.forEach(term => {return term.args[1] % 2 == 0}) : true
+                    && print(secondTerm) == print(flattenOperands(matchSecondTerm))
+                return isFactorable ? {node} : null
+            }
+        }
+    },
+
+    (node) => {
+        const {constants, coefficientMap} = getCoefficientsAndConstants(node)
+        const variables = Object.keys(coefficientMap).map(key => parse(key))
+        const coefficients = Object.keys(coefficientMap).map(key => {return coefficientMap[key]})
+
+        const firstCoef = query.getValue(coefficients[0][0])
+        const firstPoly = query.isMul(variables[0]) ? variables[0].args : [variables[0]]
+
+        const secondCoef = query.getValue(coefficients[1][0])
+
+        const thirdPoly =
+              variables[2]
+              ? query.isMul(variables[2])
+              ? variables[2].args
+              : [variables[2]]
+              : null
+
+        const thirdCoef =
+              thirdPoly ?
+              query.getValue(coefficients[2][0])
+              : constants[0]
+              ? query.getValue(constants[0])
+              : null
+
+        const firstFactors = getFactors(firstCoef)
+        const thirdFactors = getFactors(Math.abs(thirdCoef))
+
+        let combo
+
+        for (var i in firstFactors){
+            for (var j in thirdFactors) {
+                const l1 = firstFactors[i]
+                const l2 = thirdFactors[j]
```

Comment:
I have no idea what's going on here.  Can you add a comment explaining this algorithm?